torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -933,21 +935,56 @@ def _run_param_init_fn():
         self._hook_registered = False
 
     def _init_param_exec_order_wrap_policy(self, *args, **kwargs) -> None:
+        auto_wrap_policy = kwargs["auto_wrap_policy"]
+        module = kwargs["module"]
+        if isinstance(
+            auto_wrap_policy.tracing_config,
+            TracingConfig
+        ):
+            tracer = auto_wrap_policy.tracing_config.tracer
+            execution_info = _ExecutionInfo(module)
+            with _patch_tracer(
+                tracer=tracer,
+                root_module=module,
+                execution_info=execution_info,
+            ):
+                tracer.trace(module, auto_wrap_policy.tracing_config.concrete_args)
```

Comment:
what errors will user see if the trace failed?

Reply:
added an error message here

Reply:
added an error message here